src/apps/users/api/serializers.py

```diff
@@ -29,27 +38,55 @@ class Meta:
         )
 
 
-class UserCreateSerializer(serializers.ModelSerializer):
-    phone = PhoneNumberField()
+class UserRegistrationSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        required=True, validators=(UniqueValidator(queryset=User.objects.all()),)
+    )
+    password = serializers.CharField(
+        write_only=True, required=True, validators=[validate_password]
+    )
+    repeat_password = serializers.CharField(write_only=True, required=True)
 
     class Meta:
         model = User
         fields = (
             "username",
             "email",
-            "phone",
-            "first_name",
-            "last_name",
-            "type",
-            "github_link",
             "password",
+            "repeat_password",
+        )
+
+    def validate(self, attrs):
+        if attrs["password"] != attrs["repeat_password"]:
+            raise serializers.ValidationError(
+                {"password": "Password fields didn't match"}
+            )
+
+        return attrs
+
+    def create(self, validated_data):
+        user = User.objects.create(
+            username=validated_data["username"],
+            email=validated_data["email"],
+            is_active=False,
+        )
+
+        user.set_password(validated_data["password"])
+        user.save()
+
+        token = RefreshToken.for_user(user).access_token
+        verify_email_link = (
+            f"{os.getenv('SITE')}{reverse('user-verify')}?token={str(token)}"
```

Comment:
use query keyword argument instead of this param
`/api/v1/users/verify/<token>/` instead of `/api/v1/users/verify/?token=token`